test/models/payment_test.rb

```diff
@@ -1,7 +1,29 @@
 require "test_helper"
 
 class PaymentTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  
+  def refund_function
```

Comment:
- payment_result would better work as argument to this function
- you don't need any instance variables inside function